tests/trial_tests/test_trial.py

```diff
@@ -601,8 +601,30 @@ def test_report() -> None:
     with pytest.raises(TypeError):
         trial.report("foo", -1)  # type: ignore
 
+
+def test_report_step() -> None:
+    study = create_study()
+    trial = study.ask()
```

Comment:
Should I use this like existing test?

```suggestion
    trial = Trial(study, study._storage.create_new_trial(study._study_id))
```

Reply:
The existing tests seem to have been written before the `ask` feature was added. Your implementation using `ask` is fine.